Pr0gramm/Pr0gramm/Misc/StickyBadgeShowable.swift

```diff
@@ -0,0 +1,53 @@
+
+import UIKit
+import AVKit
+
+protocol StickyBadgeShowable {
+    var stickyBadgeView: UIView? { get set }
+}
+
+extension StickyBadgeShowable where Self: UIView {
+    mutating func addStickyBadge() {
+        guard stickyBadgeView == nil else { return }
+        stickyBadgeView = UIImageView(image: UIImage(systemName: "bookmark.circle.fill"))
+        stickyBadgeView?.backgroundColor = UIColor(resource: .grau)
+        stickyBadgeView?.tintColor = UIColor(resource: .sticky)
+        stickyBadgeView?.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(stickyBadgeView!)
+        stickyBadgeView?.heightAnchor.constraint(equalToConstant: 30).isActive = true
+        stickyBadgeView?.widthAnchor.constraint(equalToConstant: 30).isActive = true
+        stickyBadgeView?.rightAnchor.constraint(equalTo: rightAnchor, constant: -5).isActive = true
+        stickyBadgeView?.topAnchor.constraint(equalTo: topAnchor, constant: 5).isActive = true
+        stickyBadgeView?.transform = CGAffineTransform(scaleX: 0, y: 0)
+        stickyBadgeView?.layoutIfNeeded()
+        stickyBadgeView?.layer.cornerRadius = (stickyBadgeView?.bounds.height ?? 0) / 2
+        stickyBadgeView?.alpha = 0.5
+      
+        UIView.animate(withDuration: 0.25) { [weak self] in
```

Comment:
Hier brauchst du kein [weak self], da UIView.animate self nicht captured. Gilt z.B. auch für DispatchQueue.main.async

Reply:
ich merge es so, kannst du ja in nächsten commits wieder rausnehmen